### PR TITLE
Move next/head to Typescript

### PR DESCRIPTION
**Pull request by timneutkens**
Solves a bunch of inconsistencies in handling React elements too.

### CODE REVIEW

**Review by dav-is — APPROVED** [review]
Nice :+1:

File: package.json [hunk 1]
```diff
@@ -33,7 +33,11 @@
       "git add"
     ],
     "*.ts": [
-      "tslint -c tslint.json 'packages/**/*.ts'",
+      "tslint -c tslint.json 'packages/**/*.ts' --fix",
+      "git add"
+    ],
+    "*.tsx": [
+      "tslint -c tslint.json 'packages/**/*.ts' --fix",
```
Comment (damianfrizzi) [inline package.json]: Shouldn't this be `"tslint -c tslint.json 'packages/**/*.tsx' --fix"`?

Comment (damianfrizzi) [reply package.json]: There you go https://github.com/zeit/next.js/pull/6145

Comment (timneutkens) [reply package.json]: Merged!

**Pull request merged**